db/write_thread.cc

```diff
@@ -182,31 +188,40 @@ void WriteThread::SetState(Writer* w, uint8_t new_state) {
   }
 }
 
-WriteThreadImpl::WriteThreadImpl(uint64_t max_yield_usec,
-                                 uint64_t slow_yield_usec)
-    : WriteThread(max_yield_usec, slow_yield_usec), newest_writer_(nullptr) {}
-
-void WriteThreadImpl::LinkOne(Writer* w, bool* linked_as_leader) {
+bool WriteThread::LinkOne(Writer* w, std::atomic<Writer*>& newest_writer) {
```

Comment:
You know Google C++ style discourage non-const reference as arguments, and suggest to you pointers instead right?